service/matching/ack_manager.go

```diff
@@ -112,29 +128,43 @@
 	}
 }
 
-func (m *ackManager) completeTask(taskID int64) (ackLevel int64) {
+func (m *ackManager) completeTask(taskID int64) int64 {
 	m.Lock()
 	defer m.Unlock()
-	if completed, ok := m.outstandingTasks[taskID]; ok && !completed {
-		m.outstandingTasks[taskID] = true
-		m.backlogCounter.Dec()
+	macked, found := m.outstandingTasks.Get(taskID)
+	acked := macked.(bool)
+	if !found || acked {
+		// don't adjust ack level if nothing has changed
+		return m.ackLevel
 	}
 
 	// TODO the ack level management should be done by a dedicated coroutine
 	//  this is only a temporarily solution
+	m.outstandingTasks.Put(taskID, true)
+	m.backlogCounter.Dec()
 
-	taskIDs := maps.Keys(m.outstandingTasks)
-	util.SortSlice(taskIDs)
+	min, _ := m.outstandingTasks.Min()
+	if taskID != min {
```

Comment:
taskID here is int64 and min is interface{}.. I don't think this comparison is valid. actually I'm not sure how the compiler accepts this...